jf_agent/jf_jira/jira_download.py

```diff
@@ -505,18 +508,22 @@ def download_necessary_issues(
 # were excluded from the list of fields. Strip them out.
 def _filter_changelogs(issues, include_fields, exclude_fields):
     def _strip_history_items(items):
-        # Skip items that:
-        #  - don't have a fieldId at all (occasional internal stuff?)
-        #  - were a change of a field that's filtered out
+        # Skip items that are a change of a field that's filtered out
         for i in items:
-            if 'fieldId' not in i:
-                continue
-            if include_fields and i['fieldId'] not in include_fields:
+            field_id_field = _get_field_identifier(i)
+            if not field_id_field:
+                logger.warning(
+                    f"OJ-9084: Changlog history item with no 'fieldId' or 'field' key: {i}"
+                )
```

Comment:
Best practice in agent is to use `log_and_print` instead of `logger.warning`.

Also we probably shouldn't include `{i}` in the output, in case that contains customer-sensitive data. `{i.keys()}` should be ok though.

Reply:
:+1: